apps/bridge-dapp/src/components/Header/WalletDropdown/index.tsx

```diff
@@ -0,0 +1,145 @@
+import { Trigger as DropdownTrigger } from '@radix-ui/react-dropdown-menu';
+import { Account } from '@webb-tools/abstract-api-provider';
+import { useWebContext } from '@webb-tools/api-provider-environment';
+import { ManagedWallet, WalletConfig } from '@webb-tools/dapp-config';
+import { WebbError, WebbErrorCodes } from '@webb-tools/dapp-types';
+import {
+  ExternalLinkLine,
+  LoginBoxLineIcon,
+  WalletLineIcon,
+} from '@webb-tools/icons';
+import { useWallets } from '@webb-tools/react-hooks';
+import { WebbWeb3Provider, isViemError } from '@webb-tools/web3-api-provider';
+import {
+  Button,
+  Dropdown,
+  DropdownBody,
+  KeyValueWithButton,
+  Typography,
+  WalletButton,
+  useWebbUI,
+} from '@webb-tools/webb-ui-components';
+import { FC, useCallback, useMemo } from 'react';
+import NoteAccountAction from './NoteAccountAction';
+import NoteAccountKey from './NoteAccountKey';
+
+export const WalletDropdown: FC<{ account: Account; wallet: WalletConfig }> = ({
+  account,
+  wallet,
+}) => {
+  const { activeApi, activeChain, inactivateApi } = useWebContext();
+
+  const { notificationApi } = useWebbUI();
+
+  // Get all managed wallets
+  const { wallets } = useWallets();
+
+  const currentManagedWallet = useMemo<ManagedWallet | undefined>(() => {
+    return wallets.find((w) => w.connected);
+  }, [wallets]);
+
+  // Calculate the account explorer url
+  const accountExplorerUrl = useMemo(() => {
+    if (!activeChain?.blockExplorers) return '#';
+
+    const url = activeChain.blockExplorers.default.url;
+
+    return new URL(`/address/${account.address}`, url).toString();
+  }, [activeChain, account]);
+
+  // Funciton to switch account within the connected wallet
+  const handleSwitchAccount = useCallback(async () => {
+    // Switch account only support on web3 provider
+    if (!activeApi || !(activeApi instanceof WebbWeb3Provider)) {
+      return;
+    }
+
+    try {
+      const walletClient = activeApi.walletClient;
+
+      await walletClient.requestPermissions({ eth_accounts: {} });
+    } catch (error) {
+      let message = WebbError.from(WebbErrorCodes.SwitchAccountFailed).message;
+
+      if (isViemError(error)) {
+        message = error.shortMessage;
+      }
+
+      notificationApi({ variant: 'error', message });
+    }
+  }, [activeApi, notificationApi]);
+
+  // Disconnect function
+  // TODO: The disconnect function does not work properly
+  const handleDisconnect = useCallback(async () => {
+    if (currentManagedWallet && currentManagedWallet.canEndSession) {
```

Comment:
Consider implementing try/catch blocks within this and subsequent asynchronous functions to enhance error handling. Additionally, utilizing alerts to display messages in the UI can improve user experience by providing informative feedback.